### PR TITLE
Add ERR_BAD_CATALOG_AND_DB_ERROR error code

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [x] others

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/7785

## Problem Summary(Required) ：

error catalog in query -> ERR_BAD_CATALOG_ERROR
error catalog.db in use statement -> ERR_BAD_CATALOG_AND_DB_ERROR

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java [hunk 1]
```diff
@@ -258,9 +258,11 @@ public enum ErrorCode {
     ERR_QUERY_EXCEPTION(5077, new byte[] {'4', '2', '0', '0', '0'},
             "Query cancelled by crash of backends."),
     ERR_BAD_CATALOG_ERROR(5078, new byte[] {'4', '2', '0', '0', '0'},
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/common/ErrorCode.java:260]: can be removed now?

**Review by caneGuy — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 5 (20.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/common/ErrorCode.java|1|1|100.00%|

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by stephen-shelby** [comment]
@Mergifyio backport branch-2.3

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**